test/tactic_set.lean

```diff
@@ -0,0 +1,389 @@
+/-
+Copyright (c) 2021 Martin Zinkevich. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Zinkevich
+-/
+import tactic.set
+import tactic.ext
+import tactic.tauto
+import data.set.default
+import data.finset.default
+
+/-! The set equality lemmas from the file data.set.basic, reproven with set_taut or set_taut'.
+
+  Did not include iff lemmas nor those that had additional
+  conditions that didn't work. -/
+open set
+
+namespace set_tests
+universe v
+variables {α β: Type*} {a:α} {s t u : set α} {f g : β → set α} 
+
+lemma top_eq_univ : (⊤ : set α) = univ := by set_taut
+lemma bot_eq_empty : (⊥ : set α) = ∅ :=  by set_taut
+lemma sup_eq_union (s t : set α) : s ⊔ t = s ∪ t := by set_taut
+lemma inf_eq_inter (s t : set α) : s ⊓ t = s ∩ t := by set_taut
+lemma le_eq_subset (s t : set α) : s ≤ t = (s ⊆ t) := by set_taut
+
+
+
+
+
+theorem set_of_set {s : set α} : set_of s = s := by set_taut
+
+lemma sep_set_of {p q : α → Prop} : {a ∈ {a | p a } | q a} = {a | p a ∧ q a} := by set_taut
+
+lemma set_of_and {p q : α → Prop} : {a | p a ∧ q a} = {a | p a} ∩ {a | q a} := by set_taut
+
+lemma set_of_or {p q : α → Prop} : {a | p a ∨ q a} = {a | p a} ∪ {a | q a} := 
+ by set_taut
+
+theorem empty_def : (∅ : set α) = {x | false} :=  by set_taut
+
+theorem set_of_false : {a : α | false} = ∅ := by set_taut
+
+theorem set_of_true : {x : α | true} = univ := by set_taut
+
+theorem union_def {s₁ s₂ : set α} : s₁ ∪ s₂ = {a | a ∈ s₁ ∨ a ∈ s₂} := by set_taut
+
+theorem union_self (a : set α) : a ∪ a = a := by set_taut
+
+theorem union_empty (a : set α) : a ∪ ∅ = a := by set_taut
+
+theorem empty_union (a : set α) : ∅ ∪ a = a := by set_taut
+
+theorem union_comm (a b : set α) : a ∪ b = b ∪ a := by set_taut
+
+theorem union_comm' (a b : set α) : a ∪ b = b ∪ a := begin
+  ext x,
+  repeat { rw set.mem_union_eq },
+  tautology,
+end
```

Comment:
Could you add a comment in the code to explain what is interesting about this alternate proof? It doesn't use the `set_taut` tactic which is being tested in this file (unlike the previous proof, of the same statement).

Reply:
I think I was trying to explicitly figure out what [simp] was doing here. I don't think it helps, I'll delete.